tasks/preflight.yml

```diff
@@ -51,14 +56,41 @@
   set_fact:
     cmos_exporter_systemd_version: "{{ __systemd_version.stdout_lines[0] | regex_replace('^systemd\\s(\\d+).*$', '\\1') }}"
 
+- block:
+    - name: Get the Couchbase Cluster Info
+      no_log: true
+      uri:
+        url: http://{{ inventory_hostname }}:8091/pools/default
+        method: GET
+        user: "{{ couchbase_username }}"
+        password: "{{ couchbase_password }}"
+        timeout: 2
+      register: _results
+
+    - name: Set the Couchbase Server Version from the result of API call
+      set_fact:
+        couchbase_server_version: "{{ _results.json | community.general.json_query('nodes[?thisNode].version | [0]') }}"
```

Comment:
Does `community.general` need to be added to the requirements file? 

>This collection is shipped with the Ansible package. So if you have it installed, no more action is required.

> If you have a minimal installation (only Ansible Core installed) or you want to use the latest version of the collection along with the whole Ansible package, you need to install the collection from [Ansible Galaxy](https://galaxy.ansible.com/community/general) manually with the ansible-galaxy command-line tool:

Reply:
Roles can't rely on external collections through requirements.  Might need to look at a different way of achieving the same thing as `json_query`